workflow-ui/core-common/src/main/java/com/squareup/workflow1/ui/ViewRegistry.kt

```diff
@@ -61,24 +63,60 @@ import kotlin.reflect.KClass
  */
 @WorkflowUiExperimentalApi
 public interface ViewRegistry {
+  /**
+   * Identifies a UI factory [Entry] in a [ViewRegistry].
+   *
+   * @param renderingType the type of view model for which [factoryType] instances can build UI
+   * @param factoryType the type of the UI factory that can build UI for [renderingType]
+   */
+  public class Key<in RenderingT : Any, out FactoryT : Any>(
+    public val renderingType: KClass<in RenderingT>,
+    public val factoryType: KClass<out FactoryT>
+  ) {
+    override fun equals(other: Any?): Boolean {
+      if (this === other) return true
+      if (javaClass != other?.javaClass) return false
+
+      other as Key<*, *>
+
+      if (renderingType != other.renderingType) return false
+      return factoryType == other.factoryType
+    }
+
+    override fun hashCode(): Int {
+      var result = renderingType.hashCode()
+      result = 31 * result + factoryType.hashCode()
+      return result
+    }
+
+    override fun toString(): String {
+      return "Key(renderingType=$renderingType, factoryType=$factoryType)"
+    }
+  }
+
+  /**
+   * Implemented by a factory that can build some kind of UI for view models
+   * of type [RenderingT], and which can be listed in a [ViewRegistry]. The
+   * [Key.factoryType] field of [key] must be the type of this [Entry].
+   */
   public interface Entry<in RenderingT : Any> {
```

Comment:
It is very tempting to rename `Entry` to `UiFactory`. I avoided that in this PR to minimize surface area and migration pain. But maybe we should allow ourselves one more round of renaming before locking everything down?

Reply:
Nah.